File: src/main/java/cc/polyfrost/polyblur/config/PolyBlurConfig.java
package cc.polyfrost.polyblur.config;

import cc.polyfrost.oneconfig.config.Config;
import cc.polyfrost.oneconfig.config.annotations.Dropdown;
import cc.polyfrost.oneconfig.config.annotations.Info;
import cc.polyfrost.oneconfig.config.annotations.Slider;
import cc.polyfrost.oneconfig.config.annotations.Switch;
import cc.polyfrost.oneconfig.config.data.InfoType;
import cc.polyfrost.oneconfig.config.data.Mod;
import cc.polyfrost.oneconfig.config.data.ModType;
import cc.polyfrost.polyblur.blurs.phosphor.PhosphorBlur;

public class PolyBlurConfig extends Config {

    @Info(
            text = "Phosphor / Moulberry blur will ONLY work if either Fast Render is disabled or Force Disable Fast Render is enabled.",
            size = 2,
            type = InfoType.WARNING
    )
    private boolean agajsjg = false;

    @Switch(
            name = "Force Disable Fast Render"
    )
    public boolean forceDisableFastRender = true;

    @Dropdown(
            name = "Blur Mode",
            options = {
                    "Monkey Blur",
                    "Phosphor Blur",
                    "Moulberry Blur"
            }
    )
    public int blurMode = 1;

    @Slider(
        name = "Blur Strength",
        min = 1, max = 10,
        step = 1
    )
    public int strength = 3;

    public PolyBlurConfig() {
        super(new Mod("PolyBlur", ModType.PVP, "/polyblur_dark.svg"), "polyblur.json");
        initialize();
        addListener("strength", () -> {
            if (blurMode == 1 && enabled) {
                PhosphorBlur.reloadIntensity();
            }
        });
        addListener("blurMode", PhosphorBlur::reloadBlur);
    }
}

File: src/main/kotlin/org/polyfrost/polyblur/client/PolyBlurConfig.kt
package org.polyfrost.polyblur.client

import org.polyfrost.oneconfig.api.config.v1.KtConfig
import org.polyfrost.polyblur.PolyBlurConstants

object PolyBlurConfig : KtConfig(
    id = "${PolyBlurConstants.ID}.json",
    title = PolyBlurConstants.NAME,
    category = Category.COMBAT,
    icon = "/assets/polyblur/polyblur_dark.svg"
) {
    var isEnabled by switch(def = true, name = "Enabled", description = "Turns PolyBlur on or off.")
    var blurType by dropdown(
        options = arrayOf("Phosphor", "Unity", "Hybrid"),
        def = 2,
        name = "Blur Type",
        description = "Phosphor smears old frames together. Unity blurs along camera movement. Hybrid uses Unity for the world and Phosphor for your hand."
    )
    var phosphorMode by dropdown(
        options = arrayOf("Weighted Max", "Linear Mix", "Alpha Decay"),
        def = 1,
        name = "Phosphor Mode",
        description = "How each frame blends with the last one. Changes how the trail looks and how long it sticks around."
    )
    var strength by slider(
        min = 1f,
        max = 10f,
        def = 3f,
        name = "Blur Strength",
        description = "How strong the blur is."
    )
    var handBlurStrength by slider(
        min = 1f,
        max = 10f,
        def = 5f,
        name = "Hand Blur Strength",
        description = "How strong the blur is on your held item in Hybrid mode."
    )
    var motionBlurSamples by slider(
        min = 4f,
        max = 32f,
        def = 16f,
        name = "Motion Blur Samples",
        description = "How many samples the blur is built from. Higher is smoother but costs more FPS."
    )

    var velocityBuffer by switch(
        def = true,
        name = "Velocity Buffer",
        description = "Tracks per pixel movement for a more accurate blur. Ignored while Iris shaders are on."
    )
    var translationParallax by switch(
        def = true,
        name = "Translation Parallax",
        description = "Also blurs when you walk around, not only when you turn the camera."
    )
    var blurHand by switch(
        def = true,
        name = "Blur Hand",
        description = "Blurs your held item along with the world."
    )

    init {
        hideIf(::phosphorMode) { blurType != 0 && blurType != 2 }
        hideIf(::handBlurStrength) { blurType != 2 }
        hideIf(::blurHand) { blurType == 2 }
    }
}
